### PR TITLE
Add diagnostic log doc

### PR DESCRIPTION
**Pull request by wanlwanl**

### CODE REVIEW

File: docs/diagnostic-logs.md [hunk 1]
```diff
@@ -0,0 +1,164 @@
...
+  - [Unexpected Connection Dropping](#unexpected-connection-dropping)
+  - [Unexpected Connection Growing](#unexpected-connection-growing)
+  
+# Prerequisites
+To enable diagnostic logs, you'll need somewhere to store your log data. This tutorial uses Azure Storage and Log Analytics.
+
+* [Azure storage](https://docs.microsoft.com/en-us/azure/azure-monitor/platform/resource-logs-collect-storage) - Retains diagnostic logs for policy audit, static analysis, or backup.
+* [Log Analytics](https://docs.microsoft.com/en-us/azure/azure-monitor/platform/resource-logs-collect-workspace) - A flexible log search and analytics tool that allows for analysis of raw logs generated by an Azure resource.
+
+# Set up diagnostic logs for an Azure SignalR Service
```
Comment (chenkennt) [inline docs/diagnostic-logs.md]: Move title to be beginning?

File: docs/diagnostic-logs.md [hunk 2]
```diff
@@ -0,0 +1,164 @@
...
+Location | Location of your Azure SignalR Service
+Level | Log event level
+CallerIpAddress | IP address of your server/client
+Message | Detailed message of log event
+UserId | Identity of the user
+ConnectionId | Identity of the connection
+ConnectionType | Type of the connection. Allowed values are: `Server` \| `Client`. `Server`: connection from server side; `Client`: connection from client side
+TransportType | Transport type of the connection. Allowed values are: `Websockets` \| `ServerSentEvents` \| `LongPolling`
+
+## Trouble Shooting With Diagnostic Logs
```
Comment (chenkennt) [inline docs/diagnostic-logs.md]: Troubleshoot is a single word.

File: docs/diagnostic-logs.md [hunk 3]
```diff
@@ -0,0 +1,164 @@
...
+
+Typical issues are often about connections's unexpected quantity changes. See the next sections about how to trouble shoot.
+
+## Unexpected Connection Dropping
+
+If you encounter unexpected connections drop, firstly enable logs in service, server and client sides.
+
+If a connection disconnects, the diagnostic logs will record this disconnecting event, you will see `ConnectionAborted` or `ConnectionEnded` in `operationName`.
+
+The difference between `ConnectionAborted` and `ConnectionEnded` is that `ConnectionEnded` is a expected disconnecting which is triggered by client or server side. While the `ConnectionAborted` is usually an unexpected connection dropping event, and aborting reason will be provided in `message`.
```
Comment (chenkennt) [inline docs/diagnostic-logs.md]: *an* expected

File: docs/diagnostic-logs.md [hunk 4]
```diff
@@ -0,0 +1,164 @@
...
+The abort reasons are listed in the following table:
+
+Reason | Description
+------- | ------- 
+Connection count reaches limit | Connection count reaches limit of your current price tier. Consider scale up service unit
+Application server closed the connection | App server triggers the abortion. It can be considered as an expected abortion
+Connection ping timeout | Usually it is caused by network issue. Consider check your app server's availability from the internet
+Service reloading, please reconnect | Azure SignalR Service is reloading. Azure SignalR support auto-reconnect, you can wait until reconnected or manually reconnect to Azure SignalR Service
+Internal server transient error | Transient error occurs in Azure SignalR Service, should be auto-recovered
+Server connection dropped | Server connection drops with unknown error
```
Comment (chenkennt) [inline docs/diagnostic-logs.md]: Is there a reason that client is disconnected because server connected is closed?

Comment (chenkennt) [inline docs/diagnostic-logs.md]: What customer should do for “server connection dropped"?

File: docs/diagnostic-logs.md [hunk 5]
```diff
@@ -0,0 +1,164 @@
...
+
+## Trouble Shooting With Diagnostic Logs
+
+To trouble shoot for Azure SignalR Service, you can enable server/client side logs to capture failures. At present, Azure SiganlR Service exposes diagnostic logs, you can also enable logs for service side.
+
+When encountering connection unexpected growing or dropping situation, you can take advantage of diagnostic logs to trouble shoot.
+
+Typical issues are often about connections's unexpected quantity changes. See the next sections about how to trouble shoot.
+
+## Unexpected Connection Dropping
```
Comment (chenkennt) [inline docs/diagnostic-logs.md]: Be consistent with other headings. Only capitalize the first word.

File: docs/diagnostic-logs.md [hunk 6]
```diff
@@ -0,0 +1,164 @@
...
+Connection count reaches limit | Connection count reaches limit of your current price tier. Consider scale up service unit
+Application server closed the connection | App server triggers the abortion. It can be considered as an expected abortion
+Connection ping timeout | Usually it is caused by network issue. Consider check your app server's availability from the internet
+Service reloading, please reconnect | Azure SignalR Service is reloading. Azure SignalR support auto-reconnect, you can wait until reconnected or manually reconnect to Azure SignalR Service
+Internal server transient error | Transient error occurs in Azure SignalR Service, should be auto-recovered
+Server connection dropped | Server connection drops with unknown error
+
+## Unexpected Connection Growing
+
+To trouble shoot about unexoected connection growing, the first thing you need to do is filter out the extra connections. You can add unique test user ID to your test client connection. Then verify it in with diagnostic logs, you see more than one client connections have the same test user ID, then it is likely the client side create and establish more connections than expectation. Check your client side.
```
Comment (chenkennt) [inline docs/diagnostic-logs.md]: unexoected => unexpected

Comment (chenkennt) [reply docs/diagnostic-logs.md]: Also check multiple connections from same IP?

File: docs/diagnostic-logs.md (hunk 2 above)
Comment (chenkennt) [inline docs/diagnostic-logs.md]: Is there any troubleshooting guide for `Authorization` and `Throttling` category?

File: docs/diagnostic-logs.md (hunk 4 above)
Comment (realwanpengli) [reply docs/diagnostic-logs.md]: this one is in the reason column of client connection drops due to server connection drops

File: docs/diagnostic-logs.md [hunk 7]
```diff
@@ -0,0 +1,197 @@
+# Diagnostic Logs for Azure SignalR Service
+
+## Contents
+- [Diagnostic Logs for Azure SignalR Service](#diagnostic-logs-for-azure-signalr-service)
+  - [Contents](#contents)
```
Comment (chenkennt) [inline docs/diagnostic-logs.md]: Remove these two lines?

File: docs/diagnostic-logs.md [hunk 8]
```diff
@@ -0,0 +1,197 @@
...
+callerIpAddress | IP address of your server/client
+properties | Detailed properties related to this log event. For more detail, see [`Properties Table`](#properties-table)
+
+<a href="properties-table"></a>
+**Properties Table**
+
+Name | Description
+------- | -------
+type | Collection of the log event. Currently, we provide information about connectivity to the Azure SignalR Service. Only `ConnectivityLogs` collection is available
+collection | Type of the log event. Allowed values are: `Connection`, `Authorization` and `Throttling`
```
Comment (chenkennt) [inline docs/diagnostic-logs.md]: The descriptions seem to be wrong way around? (Why type's description is about collection and collection's description is about type?)

File: docs/diagnostic-logs.md (hunk 4 above)
Comment (realwanpengli) [reply docs/diagnostic-logs.md]: updated

File: docs/diagnostic-logs.md (hunk 6 above)
Comment (realwanpengli) [reply docs/diagnostic-logs.md]: updated

File: docs/diagnostic-logs.md (hunk 2 above)
Comment (realwanpengli) [reply docs/diagnostic-logs.md]: updated

File: docs/diagnostic-logs.md (hunk 8 above)
Comment (realwanpengli) [reply docs/diagnostic-logs.md]: No. the type is about different log types, collection is different set in the same log type

Comment (chenkennt) [reply docs/diagnostic-logs.md]: But this looks very confusing. Why type is collection of the log event and collection is type of the log event.

Comment (wanlwanl) [reply docs/diagnostic-logs.md]: Oh, I see...

File: docs/diagnostic-logs.md [hunk 9]
```diff
@@ -0,0 +1,197 @@
...
+
+### Troubleshooting with diagnostic logs
+
+To troubleshoot for Azure SignalR Service, you can enable server/client side logs to capture failures. At present, Azure SiganlR Service exposes diagnostic logs, you can also enable logs for service side.
+
+When encountering connection unexpected growing or dropping situation, you can take advantage of diagnostic logs to troubleshoot.
+
+Typical issues are often about connections's unexpected quantity changes, connections reach connection limits and authorization failure. See the next sections about how to troubleshoot.
+
+#### Unexpected connections' number changes
```
Comment (chenkennt) [inline docs/diagnostic-logs.md]: connection number

File: docs/diagnostic-logs.md [hunk 10]
```diff
@@ -0,0 +1,197 @@
...
+
+#### Authorization failure
+
+If you get 401 Unauthorized returned for client requests, check your diagnostic logs. If you encounter `Failed to validate audience. Expected Audiences: <valid audience>. Actual Audiences: <actual audience>`, it means your all audiences in your access token is invalid. Try to use the valid audiences suggested in the log.
+
+
+#### Throttling
+
+If you find that you cannot establish SignalR client connections to Azure SignalR Service, check your diagnostic logs. If you encounter `Connection count reaches limit` in diagnostic log, you establish too much connections to SignalR Service, which reach the connection count limit. Consider scaling up your SignalR Service. If you encounter `Message count reaches limit` in diagnostic log, it means you use free tier, and you use up the quota of messages. If you want to send more messages, consider changing your SignalR Service to standard tier to send additional messages. For more details, see [Azure SignalR Service Pricing](https://azure.microsoft.com/en-us/pricing/details/signalr-service/).
```
Comment (chenkennt) [inline docs/diagnostic-logs.md]: too ~much~ many connections

**Pull request merged**